test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/QueryExecution/QueryExecutionServiceTests.cs

```diff
@@ -0,0 +1,116 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts;
+using Microsoft.SqlTools.ServiceLayer.QueryExecution.Contracts.ExecuteRequests;
+using Moq;
+using NUnit.Framework;
+
+namespace Microsoft.SqlTools.ServiceLayer.IntegrationTests.QueryExecution
+{
+    /// <summary>
+    /// Tests for the ServiceHost Query Execution Service tests that require a live database connection
+    /// </summary>
+    public class QueryExecutionServiceTests
+    {
+        
+
+
+        [Test]
+        public async Task QueryExecutionSessionOptionsPersistAcrossQueries()
+        {
+            // Establish a new connection
+            ConnectionService.Instance.OwnerToConnectionMap.Clear();
+            ConnectionInfo connectionInfo = LiveConnectionHelper.InitLiveConnectionInfo().ConnectionInfo;
+
+            { // check the initial default transaction level
+                var queryResult = await ExecuteAndVerifyQuery(
+                    "SELECT transaction_isolation_level FROM sys.dm_exec_sessions WHERE session_id=@@SPID;",
+                    connectionInfo.OwnerUri);
+
+                Assert.AreEqual(queryResult.RowCount, 1);
+                Int16 transactionLevel = (Int16)queryResult.Rows[0][0].RawObject;
+                Assert.AreEqual(1, transactionLevel);
+            }
+
+            { // change the transaction level and run a query
+                var queryResult = await ExecuteAndVerifyQuery(
+                    "SET TRANSACTION ISOLATION LEVEL SERIALIZABLE; SELECT transaction_isolation_level FROM sys.dm_exec_sessions WHERE session_id=@@SPID;",
+                    connectionInfo.OwnerUri);
+
+                Assert.AreEqual(queryResult.RowCount, 1);
+                Int16 transactionLevel = (Int16)queryResult.Rows[0][0].RawObject;
+                Assert.AreEqual(4, transactionLevel);
+            }
+
+            { // rerun the query without setting execution option and confirm previous option persists
+                var queryResult = await ExecuteAndVerifyQuery(
+                    "SELECT transaction_isolation_level FROM sys.dm_exec_sessions WHERE session_id=@@SPID;",
+                    connectionInfo.OwnerUri);
+
+                Assert.AreEqual(queryResult.RowCount, 1);
+                Int16 transactionLevel = (Int16)queryResult.Rows[0][0].RawObject;
+                Assert.AreEqual(4, transactionLevel);
+            }
```

Comment:
Can we also have a test case covering doc renaming scenario if it can fit into existing tests easily?